### PR TITLE
Make `source.hash` non-optional, remove `metadata-only`

### PR DESCRIPTION
**Pull request by cmichi**

### CODE REVIEW

File: src/cmd/metadata.rs [hunk 1]
```diff
@@ -226,11 +216,11 @@ impl GenerateMetadataCommand {
         let user = self.crate_metadata.user.clone().map(User::new);
 
         Ok(ExtendedMetadataResult {
-            dest_wasm,
+            dest_wasm: Some(dest_wasm),
```
Comment (Robbepop) [inline src/cmd/metadata.rs:219]: Useful to be an `Option`?

Comment (cmichi) [reply src/cmd/metadata.rs:219]: So this is mostly used for `cargo contract check`, which should never result in a `dest_wasm` in the target directory. So for the `GenerationResult` object `dest_wasm` was made an `Option`. There is also an assert for `is_none()` in the `cargo contract check` logic.

Comment (Robbepop) [reply src/cmd/metadata.rs:219]: okay cool, thanks!

**Review by Robbepop — APPROVED** [review]
LGTM!

**Review by ascjones — APPROVED** [review]

**Pull request merged**